docs/src/components/NavbarItems/DocsDropdown.tsx

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import { useLocation } from '@docusaurus/router';
+import DropdownNavbarItem from '@theme/NavbarItem/DropdownNavbarItem';
+import styles from './DocsDropdown.module.css';
+
+type Section = 'genai' | 'ml' | 'default';
+
+interface DocsDropdownProps {
+  mobile?: boolean;
+  position?: 'left' | 'right';
+  items?: any[];
+  label?: string;
+  [key: string]: any;
+}
+
+export default function DocsDropdown({ 
+  mobile, 
+  items: configItems, 
+  label: configLabel,
+  ...props 
+}: DocsDropdownProps): JSX.Element {
+  const location = useLocation();
+  
+  const getCurrentSection = (): Section => {
+    const path = location.pathname;
+    if (path.includes('/genai') || path.startsWith('/genai')) {
+      return 'genai';
+    } else if (path.includes('/ml') || path.startsWith('/ml')) {
+      return 'ml';
+    }
+    return 'default';
+  };
+
+  const currentSection = getCurrentSection();
+
+  const getLabel = (): string => {
+    switch (currentSection) {
+      case 'genai':
+        return 'GenAI Docs';
+      case 'ml':
+        return 'ML Docs';
+      default:
+        return configLabel || 'Documentation';
+    }
+  };
```

Comment:
the label we pass to `<DropdownNavbarItem>` can be a react element, so let's just construct the circle + label element here:

```suggestion
  const getLabel = (): JSX.Element => {
    let color;
    let text = 'Documentation';

    if (currentSection === 'genai') {
      color = 'var(--genai-color-primary)';
      text = 'GenAI Docs';
    } else if (currentSection === 'ml') {
      color = 'var(--ifm-color-primary)';
      text = 'ML Docs';
    }

    return (
      <div style={{ display: 'flex', gap: 8, alignItems: 'center', }}>
        {color && (
          <div
            className={styles.dropdownCircle}
            style={{ width: 10, height: 10, backgroundColor: color, borderRadius: 4 }}
          />
        )}
        {text}
      </div>
    );
  };
```